CDP4WebServices.API/Services/BusinessLogic/Implementation/ChainOfRdlComputationService.cs

```diff
@@ -0,0 +1,187 @@
+
+        /// <summary>
+        /// Gets the identifiers of the chain of <see cref="SiteReferenceDataLibrary"/> dependencies for a list of <see cref="EngineeringModelSetup"/>s
+        /// </summary>
+        /// <param name="engineeringModelSetups">
+        /// The <see cref="EngineeringModelSetup"/> for which the unique identifiers of the chain of <see cref="SiteReferenceDataLibrary"/> needs to be determined
+        /// </param>
+        /// <param name="transaction">
+        /// The current transaction to the datastore
+        /// </param>
+        /// <returns>
+        /// The unique identifiers of the <see cref="SiteReferenceDataLibrary"/> dependency of the provided <see cref="IEnumerable{EngineeringModelSetup}"/>
+        /// </returns>
+        public IEnumerable<Guid> QueryReferenceDataLibraryDependency(NpgsqlTransaction transaction, IEnumerable<EngineeringModelSetup> engineeringModelSetups)
+        {
+            if (this.cachedModelReferenceDataLibraries == null || !this.cachedModelReferenceDataLibraries.Any())
+            {
+                Logger.Debug("Retrieving the ModelReferenceDataLibrary objects from the cached tables in the datastore");
+
+                var modelReferenceDataLibraries = this.ModelReferenceDataLibraryDao.Read(transaction, SiteDirectoryPartition, null, true);
+
+                this.cachedModelReferenceDataLibraries = new List<ModelReferenceDataLibrary>();
+                
+                this.cachedModelReferenceDataLibraries.AddRange(modelReferenceDataLibraries);
+            }
+
+            if (this.cachedSiteReferenceDataLibraries == null || !this.cachedSiteReferenceDataLibraries.Any())
+            {
+                Logger.Debug("Retrieving the SiteReferenceDataLibrary objects from the cached tables in the datastore");
+
+                var siteReferenceDataLibraries = this.SiteReferenceDataLibraryDao.Read(transaction, SiteDirectoryPartition, null, true);
+
+                this.cachedSiteReferenceDataLibraries = new List<SiteReferenceDataLibrary>();
+
+                this.cachedSiteReferenceDataLibraries.AddRange(siteReferenceDataLibraries);
+            }
+
+            var result = new HashSet<Guid>();
+
+            foreach (var engineeringModelSetup in engineeringModelSetups)
+            {
+                if (engineeringModelSetup.RequiredRdl.Count > 1)
+                {
+                    Logger.Warn($"The EngineeringModelSetup { engineeringModelSetup.Iid } has more than 1 required rdl, this is not allowed, this EngineeringModelSetup is ignored");
+                    continue;
+                }
+
+                var modelReferenceDataLibararyIid = engineeringModelSetup.RequiredRdl.SingleOrDefault();
+                if (modelReferenceDataLibararyIid == Guid.Empty)
+                {
+                    Logger.Warn($"The EngineeringModelSetup { engineeringModelSetup.Iid } has more than 1 required rdl, this is not allowed, this EngineeringModelSetup is ignored");
```

Comment:
I think the error text is wrong for this check.

Reply:
yup doesnt seem right

Reply:
fixed